src/shared_modules/utils/tests/stringHelper_test.cpp

```diff
@@ -170,4 +170,29 @@ TEST_F(StringUtilsTest, CheckMultiReplacement)
     const auto retVal { Utils::replaceAll(string_base, "  ", " ") };
     EXPECT_EQ(string_base, "hello world");
     EXPECT_TRUE(retVal);
+}
+
```

Comment:
Maybe you should add a specific UT checking the "\n" case which is the one that, currently, is being used.